File: docker/iml-timer/start-stratagem-scan.py
#!/usr/bin/env python

# Copyright (c) 2020 DDN. All rights reserved.
# Use of this source code is governed by a MIT-style
# license that can be found in the LICENSE file.

import requests
import os, sys, getopt

args = sys.argv[1:]
short_opts = "f:r:p:"
long_opts = ["filesystem=", "report=", "purge="]

url = os.getenv("SERVER_HTTP_URL")
api_user = os.getenv("API_USER")
api_key = os.getenv("API_KEY")

try:
    arguments, values = getopt.getopt(args, short_opts, long_opts)
except getopt.error as err:
    print(str(err))
    sys.exit(2)

report_duration = None
purge_duration = None
for opt, val in arguments:
    if opt in ("-f", "--filesystem"):
        filesystem = val
    elif opt in ("-r", "--report"):
        report_duration = val
    elif opt in ("-p", "--purge"):
        purge_duration = val

post_data = {"filesystem": filesystem, "report_duration": report_duration, "purge_duration": purge_duration}
s = requests.Session()
s.headers.update({"AUTHORIZATION": "ApiKey {}:{}".format(api_user, api_key)})
response = s.post("{}/api/run_stratagem/".format(url), json=post_data, verify=False)

if not response.ok:
    raise Exception(response.reason)

File: iml-timer/src/bin/start-stratagem-scan.rs
// Copyright (c) 2020 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

use iml_manager_client::{get_client, post};
use structopt::StructOpt;
use tracing_subscriber::{fmt::Subscriber, EnvFilter};

#[derive(Debug, StructOpt)]
#[structopt(name = "start-stratagem-scan")]
/// The start-stratagem-scan API
struct App {
    #[structopt(short = "f", long = "filesystem")]
    filesystem: String,

    #[structopt(short = "r", long = "report")]
    report: Option<String>,

    #[structopt(short = "p", long = "purge")]
    purge: Option<String>,
}

#[derive(serde::Serialize)]
struct StratagemData {
    filesystem: String,
    report_duration: Option<String>,
    purge_duration: Option<String>,
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let subscriber = Subscriber::builder()
        .with_env_filter(EnvFilter::from_default_env())
        .finish();

    tracing::subscriber::set_global_default(subscriber).unwrap();

    let opts = App::from_args();

    let post_data = StratagemData {
        filesystem: opts.filesystem,
        report_duration: opts.report,
        purge_duration: opts.purge,
    };

    post(get_client()?, "run_stratagem", &post_data)
        .await?
        .error_for_status()?;

    tracing::info!("Sent request to run stratagem scan.");

    Ok(())
}
